bokehjs/test/integration/regressions.ts

```diff
@@ -4607,4 +4607,22 @@ describe("Bug", () => {
       expect(obj.value).to.not.be.equal(null)
     })
   })
+
+  describe("in issue #14520", () => {
+    it("doesn't allow BoxAnnotation to support categorical coordinates", async () => {
+      const p = fig([300, 200], {y_range: ["A", "B", "C", "D", "E", "F"]})
+
+      const box = new BoxAnnotation({bottom: "B", top: "D", fill_alpha: 0.2, fill_color: "green"})
+      p.add_layout(box)
+
+      p.scatter({
+        x: [0, 10, 20, 30, 40, 50],
+        y: ["A", "B", "C", "D", "E", "F"],
+        marker: "circle",
+        size: 10,
+      })
+
+      await display(p)
+    })
+  })
```

Comment:
Is this a regression? AFAIK it has never worked previously

Reply:
I guess my larger meta-issue is that `regression.ts` is approaching 5kLOC and is just becoming one giant unstructured dumping ground. When possible I really prefer tests be grouped by the module or API they exercise or address. Whether that means duplicating the lib layout under a `regressions` dir, or having tests under `integration` be easily tag-able and query-able somehow wrt to specific bugs they cover, I don't know. But we are nearing the point where some kind of organization needs to be applied, IMO.